bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniDrugOrderServiceImpl.java

```diff
@@ -167,6 +178,110 @@ public List<Order> getAllDrugOrders(String patientUuid, String patientProgramUui
+        for(BahmniDrugOrder drugOrder : drugOrderList) {
+            BahmniDrugOrder foundDrugOrder = mergedDrugOrderMap.entrySet().stream()
+                    .map(x -> x.getKey())
+                    .filter( existingOrder ->
+                            areValuesEqual(existingOrder.getDrugNonCoded(), drugOrder.getDrugNonCoded()) &&
+                                    (existingOrder.getDrug()!=null && drugOrder.getDrug()!=null &&
+                                            areValuesEqual(existingOrder.getDrug().getUuid(), drugOrder.getDrug().getUuid())) &&
+                                    areValuesEqual(existingOrder.getInstructions(), drugOrder.getInstructions()) &&
+                                    compareDosingInstruction(existingOrder.getDosingInstructions(), drugOrder.getDosingInstructions()) &&
+                                    areValuesEqual(existingOrder.getDosingInstructions().getRoute(), drugOrder.getDosingInstructions().getRoute()) &&
+                                    areValuesEqual(existingOrder.getDosingInstructions().getAdministrationInstructions(), drugOrder.getDosingInstructions().getAdministrationInstructions()) &&
+                                    areValuesEqual(existingOrder.getDosingInstructions().getAsNeeded(), drugOrder.getDosingInstructions().getAsNeeded()) &&
+                                    areValuesEqual(existingOrder.getDateStopped(), drugOrder.getDateStopped()) &&
+                                    getDateDifferenceInDays(existingOrder.getEffectiveStopDate(), drugOrder.getEffectiveStartDate()) <= 1.0 )
+                    .findFirst()
+                    .orElse(null);
+            if (foundDrugOrder!=null) {
+                mergedDrugOrderMap.put(foundDrugOrder, mergedDrugOrderMap.get(foundDrugOrder)+drugOrder.getDuration());
+            } else {
+                mergedDrugOrderMap.put(drugOrder, drugOrder.getDuration());
+            }
+        }
+        return mergedDrugOrderMap;
+    }
+
+    @Override
+    public String getPrescriptionAsString(Map<BahmniDrugOrder, Integer> drugOrderDurationMap, Locale locale) {
+        String prescriptionString = "";
+        int counter = 1;
+        for (Map.Entry<BahmniDrugOrder, Integer> entry : drugOrderDurationMap.entrySet()) {
+            prescriptionString += counter++ + ". " + getDrugOrderString(entry.getKey(), drugOrderDurationMap.get(entry.getKey()), locale) + "\n";
+        }
+        return prescriptionString;
+    }
+
+    @Override
+    public String getAllProviderAsString(List<BahmniDrugOrder> drugOrders) {
+        Set providerSet = new LinkedHashSet();
+        for (BahmniDrugOrder drugOrder : drugOrders) {
+            providerSet.add("Dr " + drugOrder.getProvider().getName());
+        }
+        return StringUtils.collectionToCommaDelimitedString(providerSet);
+    }
+
+    private String getDrugOrderString(BahmniDrugOrder drugOrder, Integer duration, Locale locale) {
```

Comment:
rename to getDrugOrderAsString() or getTextualDrugOrder() ?

Reply:
renamed to getDrugOrderAsString()